src/components/PopUp.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import { View, StyleSheet, ViewStyle, Modal, Alert } from "react-native";
+import type { StyleProp } from "react-native";
+import { PRIMARY_COLOR, TEXT_COLOR } from "../util/colors";
+import { Button } from "./Button";
+
+export interface ModalProps {
+	text: string;
+	accessibilityLabel: string;
+	accessibilityHint?: string;
+	style?: StyleProp<ViewStyle>;
+}
+
+export const PopUp: React.FC<ModalProps> = ({
+	text,
+	accessibilityLabel,
+	accessibilityHint,
+	style,
+	children,
+}) => {
+	const [modalVisible, setModalVisible] = useState(false);
+	return (
+		<View
+			accessibilityLabel="A Pop Up"
+			accessibilityHint="A Pop Up"
+			style={StyleSheet.compose(
+				{
+					backgroundColor: PRIMARY_COLOR,
+				},
+				style
+			)}
+		>
+			<Modal
+				animationType="none"
+				// TODO: setting 'transparent' to false will make the background disappear; we may want this for our users
+				// so that there are less distractions and they can focus on a single element. Test it out and lmk what you guys think
```

Comment:
tried it out, and I'm not too big a fan on the background disappearing, I think it looks too much like a glitch with the app or something. if we could get the background to be darkened so that the background elements are only a little visible but are still clearly there, I think that would be more effective at letting the user know that the background elements are not in focus right now.